### DISCUSSION TITLE
Clear or reload cached classes at runtime?

### DISCUSSION THREAD

**Discussion by Mikhinja** [pythonnet/pythonnet#1725] [Q&A] [answered]
I have an add-in into a larger application that runs python scripts, using python.NET.

I have noticed that if I modify only the python script I am running everything is ok.
But if I run the script once, and after that I modify one of the files imported, and run the script again, then the changes in the imported class are not applied until I restart the whole application.

I assume there is a class cache somewhere that causes this behavior.
So my question is, is there a way to force at runtime a clear, or refresh, or reload of cached objects or classes?

**Comment by filmor** [comment, marked as answer]
This is not a Python.NET but a .NET problem. You will need to use AppDomains in .NET Framework or Mono and [AssemblyLoadContext](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.loader.assemblyloadcontext) in .NET Core (or 5-6) to be able to reload DLLs.
